apps/tests/integration/integration_test.go

```diff
@@ -480,3 +492,109 @@ func TestAccountFromCache(t *testing.T) {
 	}
 
 }
+
+type urlModifierTransport struct {
+	base       http.RoundTripper
+	modifyFunc func(*http.Request)
+}
+
+// RoundTrip implements the http.RoundTripper interface
+func (t *urlModifierTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	// Modifying the original resquest to have the updated URL
+	if t.modifyFunc != nil {
+		t.modifyFunc(req)
+	}
+	return t.base.RoundTrip(req)
+}
+
+func TestAcquireMSITokenExchangeForESTSToken(t *testing.T) {
+	labC, err := newLabClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+	baseUrl := "https://service.msidlab.com/"
+	resource := "api://azureadtokenexchange"
+
+	response, err := labC.getLabResponse(baseUrl + "EnvironmentVariables?resource=WebApp")
+	if err != nil {
+		t.Fatalf("Failed to get resource env variable: %v", err)
+	}
+	var result map[string]string
+	err = json.Unmarshal([]byte(response), &result)
+	if err != nil {
+		t.Fatalf("Failed to unmarshal response: %v", err)
+	}
+
+	for key, value := range result {
+		if key == "IDENTITY_ENDPOINT" {
+			value = "https://service.msidlab.com/MSIToken?azureresource=WebApp&uri=" + value
+		}
+		os.Setenv(key, value)
+		defer os.Unsetenv(key)
```

Comment:
`t.Setenv` is better because it restores the pre-test value i.e., it prevents side-effects on other tests

Reply:
Updated